src/Form/CitizenProjectCommentCommandType.php

```diff
@@ -3,18 +3,45 @@
 namespace AppBundle\Form;
 
 use AppBundle\CitizenProject\CitizenProjectCommentCommand;
+use AppBundle\CitizenProject\CitizenProjectPermissions;
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
 use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
 use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
 
 class CitizenProjectCommentCommandType extends AbstractType
 {
+    private $authorizationChecker;
+
+    public function __construct(AuthorizationCheckerInterface $authorizationChecker)
+    {
+        $this->authorizationChecker = $authorizationChecker;
+    }
+
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
             ->add('content', TextType::class)
         ;
+
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, function (FormEvent $event) {
```

Comment:
If this form is used as root form please remove this listener, `$data->getCitizenProject()` might break if there is no initial data.

Please add something like:
```php
$data = $builder->getData();

if ($data instanceof CitizenProjectCommentCommand) {
    $builder
        ->add('sendMail', CheckboxType::class, [
            'required' => false,
            'label' => 'Envoyer aussi par e-mail',
            'attr' => ['class' => 'form__checkbox form__checkbox--large'],
        ])
    ;
}
```

Reply:
Done

Reply:
You did not remove the listener.

Reply:
You still have to remove this listener.

Reply:
We need this listener because only hosts can send mail while adding a comment.

Reply:
See https://github.com/EnMarche/en-marche.fr/pull/1919/commits/44c680ade829c39a5b26b8a1ba9b96d5f299e912.